### PR TITLE
[all] Promote `component.UseLocalHostAsDefaultHost` feature gate to beta

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** <Describe what has changed.>

Counterpart to open-telemetry/opentelemetry-collector/pull/10352.

**Link to tracking Issue:** open-telemetry/opentelemetry-collector/issues/8510

### CODE REVIEW

**Review by jpkrohling — APPROVED** [review]

File: .chloggen/mx-psi_localhost-fg.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    - receiver/influxdb
+    - receiver/jaeger
+    - receiver/loki
+    - receiver/opencensus
+    - receiver/sapm
+    - receiver/signalfx
+    - receiver/skywalking
+    - receiver/splunk_hec
+    - receiver/zipkin
+    - receiver/zookeeper
```
Comment (ChrsMark) [inline .chloggen/mx-psi_localhost-fg.yaml:35]: I see some extra components' tests have been updated as well. Should those components be listed here as well?

Comment (mx-psi) [reply .chloggen/mx-psi_localhost-fg.yaml:35]: The changes in other components are to set the endpoints explicitly for the health check extension and the OTLP receiver, but don't really change the components themselves. 


I took this list from https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/f12f0adc54de5e4a9e08b6697e182a0d653d05ad/.chloggen/mx-psi_internal-localhostgate.yaml, but as a sanity check, I grepped for the function `localhostgate.EndpointForPort`:

```
❯ rg localhostgate\.EndpointForPort -tgo --glob '!*_test.go' -l | sort
extension/awsproxy/factory.go
extension/healthcheckextension/factory.go
extension/healthcheckv2extension/factory.go
extension/jaegerremotesampling/factory.go
internal/aws/proxy/cfg.go
processor/remotetapprocessor/config.go
receiver/awsfirehosereceiver/factory.go
receiver/awsxrayreceiver/factory.go
receiver/influxdbreceiver/factory.go
receiver/jaegerreceiver/factory.go
receiver/lokireceiver/factory.go
receiver/opencensusreceiver/factory.go
receiver/sapmreceiver/factory.go
receiver/signalfxreceiver/factory.go
receiver/skywalkingreceiver/factory.go
receiver/splunkhecreceiver/factory.go
receiver/zipkinreceiver/factory.go
receiver/zookeeperreceiver/factory.go
```

The component list is the same as in the release note with the exception of `healtcheckv2extension`, so I will add that one even if it's in development

**Review by ChrsMark — APPROVED** [review]

**Review by songy23 — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Review by MovieStoreGuy — APPROVED** [review]

**Review by crobert-1 — COMMENTED** [review]
Did some basic grepping of `0.0.0.0`, below are references that are in the components referenced in the CHANGELOG.

`internal/aws/proxy`: A couple tests are using `0.0.0.0` in their port binding, I'm not familiar enough with the logic to know how impactful this is to update to `localhost`.

`receiver/awsxray`: The README has lots of references `0.0.0.0` being the default address, I assume these should be updated now since `localhost` will be the default?

`receiver/jaeger`: The README has lots of references to `0.0.0.0` being the default address as well that should probably be updated. There are also a lot of tests still using `0.0.0.0`.

`receiver/loki`: `0.0.0.0` references in README. Also, in its `factory.go`, there are two default endpoint variables referencing `0.0.0.0`. One of these variables is unused, the other is only used in a test.
```
	defaultGRPCBindEndpoint = "0.0.0.0:3600"
	defaultHTTPBindEndpoint = "0.0.0.0:3500"
```

`receiver/sapm`: References in README (along with feature gate reference that needs updated since it's enabled by default), and a few different test files.

**Comment by mx-psi** [comment]
@crobert-1 I don't feel the tests are something we need to change (if anybody disagrees, let me know), but I will open a PR to update the documentation, thank you for checking this!

**Comment by crobert-1** [comment]
> @crobert-1 I don't feel the tests are something we need to change (if anybody disagrees, let me know)

Fair point, they're especially still relevant while the feature gate can be disabled. I believe they'll need to be updated once it's graduated to stable though, unless I'm mistaken.

**Comment by mx-psi** [comment]
@crobert-1 Updated READMEs, can you double check?

**Review by crobert-1 — APPROVED** [review]
LGTM, thanks for making all of the documentation updates as well!

**Pull request merged**